datanode/entities/pagination.go

```diff
@@ -123,7 +126,7 @@ func CursorPaginationFromProto(cp *v2.Pagination) (CursorPagination, error) {
 	}
 
 	if cp.First != nil {
-		if *cp.First < 0 || *cp.First > defaultPageSize {
+		if *cp.First < 0 || *cp.First > maxPageSize {
 			return CursorPagination{}, ErrCursorOverflow
```

Comment:
Can we change the error to not be `negative pagination value` and say something useful about it no being in range `1-5000`?

Reply:
Changed that to be the message for `ErrCursorOverflow`